### ISSUE TITLE
`not-an-iterable` error for perfectly valid asyncio.couroutine decorated code

### ISSUE THREAD

**Issue by RemiCardona** [PyCQA/pylint#996]
### Steps to reproduce

``` python
import asyncio

@asyncio.coroutine
def coroutine_function_return_none():
    # both return statements trigger the bug
    return
    #return None

@asyncio.coroutine
def coroutine_function_return_object():
    return 12

@asyncio.coroutine
def coroutine_function_return_future():
    return asyncio.Future()

@asyncio.coroutine
def coroutine_function_pass():
    pass

@asyncio.coroutine
def coroutine_generator():
    yield

@asyncio.coroutine
def main(self):
    yield from coroutine_function_return_none()
    yield from coroutine_function_return_object()
    yield from coroutine_function_return_future()
    yield from coroutine_function_pass()
    yield from coroutine_generator()
```
### Current behavior

```
async_testcase.py:27: [E1133(not-an-iterable), main] Non-iterable value coroutine_function_return_none() is used in an iterating context
async_testcase.py:28: [E1133(not-an-iterable), main] Non-iterable value coroutine_function_return_object() is used in an iterating context
```

A couple of notes:
- I can't explain why `pass` actually works and a plain `return` doesn't…
- Looking at `checkers.typecheck.IterableChecker` and `checkers.utils.is_iterable`, I see there's no support for `asyncio.coroutine`, just for `yield from`, is that correct ?
### Expected behavior

A global evaluation of 10.0/10. No less ;)
### pylint --version output

```
pylint 1.6.1, 
astroid 1.4.7
Python 3.4.2 (default, Oct  8 2014, 10:45:20) 
[GCC 4.9.1]
```
